app/services/stripe/invoice_handler.rb

```diff
@@ -0,0 +1,36 @@
+class InvoiceHandler
+  def initialize(event)
+    @invoice = event.data.object
+  end
+
+  def create_invoice!
+    begin
+    	Invoice.create!(
+        brand:              get_brand,
+        stripe_invoice_id:  @invoice.id,
+        amount:             @invoice.amount_due,
+        data:               @invoice,
+        paid_at:            Time.now.utc
+      )
+    rescue Stripe::InvalidRequestError => e
```

Comment:
Once refactor to this to not hit Stripe again to find out the brand that the event/invoice is tied to, we shouldn't need to rescue here.